lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -129,25 +129,22 @@ class QcFlagService {
      * @throws {BadParameterError}
      */
     _prepareQcFlagPeriod(timestamps, targetRun) {
-        const from = timestamps.from ?? targetRun.startTime ?? null;
-        const to = timestamps.to ?? targetRun.endTime ?? null;
+        const { timeTrgStart, timeO2Start, timeTrgEnd, timeO2End } = targetRun;
+        const runStartTime = timeTrgStart?.getTime() ?? timeO2Start?.getTime() ?? null;
+        const runEndTime = timeTrgEnd?.getTime() ?? timeO2End?.getTime() ?? null;
 
-        if (from === null || to === null) {
-            if (!from && !to) {
-                return { from: null, to: null };
-            } else {
-                throw new BadParameterError('Only null QC flag timestamps are accepted as run.startTime or run.endTime is missing');
-            }
-        }
+        const from = timestamps.from ?? runStartTime ?? null;
+        const to = timestamps.to ?? runEndTime ?? null;
 
-        if (from >= to) {
+        if (from && to && from >= to) {
             throw new BadParameterError('Parameter "to" timestamp must be greater than "from" timestamp');
         }
-
-        if (from < targetRun.startTime || targetRun.endTime < to) {
-            // eslint-disable-next-line max-len
-            throw new BadParameterError(`Given QC flag period (${from}, ${to}) is out of run (${targetRun.startTime}, ${targetRun.endTime}) period`);
+        const isFromOutOfRange = from && (runStartTime && from < runStartTime || runEndTime && runEndTime <= from);
+        const isToOutOfRange = to && (runStartTime && to <= runStartTime || runEndTime && runEndTime < to);
```

Comment:
I think that here, check is simply 
`(runStartTime && timestamp < runStartTime) || (runEndTime && runEndTime < timestamp)`
Both check are strict I suppose? In that case, maybe a function `isTimestampOutOfRange` would be good, can be put just before call (in the `_prepareQcFlagPeriod`)

Reply:
No, they are not strict, `newFlag.from` starting at `run end` is incorrect, and no, it's not so simple because any of those timestamps can be null